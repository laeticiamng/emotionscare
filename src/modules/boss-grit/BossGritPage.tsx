// @ts-nocheck
"use client";
import React from "react";
import PageHeader from "@/components/ui/PageHeader";
import { Card } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import { ProgressBar } from "@/ui/ProgressBar";
import { useTimer } from "@/ui/hooks/useTimer";
import { loadTasks, upsertTask, removeTask, GritTask } from "@/lib/boss-grit/store";
import { ff } from "@/lib/flags/ff";
import { useSound } from "@/ui/hooks/useSound";          // si P5 dispo (sinon null)
import { recordEvent } from "@/lib/scores/events";    // si P6 dispo (sinon no-op)

const PRESETS: { label: string; ms: number }[] = [
  { label: "90 s",   ms: 90_000 },
  { label: "3 min",  ms: 180_000 },
  { label: "5 min",  ms: 300_000 },
  { label: "10 min", ms: 600_000 }
];

export default function BossGritPage() {
  const [selMs, setSelMs] = React.useState<number>(180_000);
  const [tasks, setTasks] = React.useState<GritTask[]>([]);
  const [newTask, setNewTask] = React.useState("");

  const t = useTimer(selMs);
  const reduced = typeof window !== "undefined" && window.matchMedia?.("(prefers-reduced-motion: reduce)").matches;
  const useNewAudio = ff?.("new-audio-engine") ?? false;
  const beep = useNewAudio ? (useSound?.("/audio/tick.mp3") ?? null) : null;

  React.useEffect(() => { setTasks(loadTasks()); }, []);

  React.useEffect(() => {
    if (t.done) {
      try {
        beep?.play?.(); // bip de fin
        if ("vibrate" in navigator && !reduced) navigator.vibrate?.(12);
        recordEvent?.({
          module: "boss-grit",
          startedAt: new Date(Date.now() - t.totalMs).toISOString(),
          endedAt: new Date().toISOString(),
          durationSec: Math.round(t.totalMs / 1000),
          score: 2 + Math.round(t.totalMs / 120000), // proxy: 90s->3, 3min->4, etc.
          meta: {
            tasksChecked: tasks.filter(x => x.done).length,
            totalTasks: tasks.length
          }
        });
      } catch {}
    }
  }, [t.done]);

  function toggleTask(id: string) {
    const next = tasks.map(x => x.id === id ? { ...x, done: !x.done } : x);
    setTasks(next); next.forEach(upsertTask);
  }
  function addTask() {
    const s = newTask.trim(); if (!s) return;
    const tsk = { id: crypto.randomUUID?.() || String(Date.now()), label: s, done: false };
    upsertTask(tsk);
    setTasks(loadTasks());
    setNewTask("");
  }
  function delTask(id: string) {
    removeTask(id);
    setTasks(loadTasks());
  }

  return (
    <main aria-label="Boss Grit">
      <PageHeader title="Boss Grit" subtitle="Micro-sessions + 3 actions pour passer à l’acte" />
      <Card>
        <section style={{ display:"grid", gap: 12 }}>
          <div style={{ display:"flex", flexWrap:"wrap", gap: 8 }}>
            {PRESETS.map(p => (
              <Button key={p.ms} onClick={() => { setSelMs(p.ms); if (t.running) t.reset(); }}>
                {p.label}
              </Button>
            ))}
          </div>

          <div style={{ display:"flex", alignItems:"center", gap: 12 }}>
            <strong>{fmtTime(t.elapsed)} / {fmtTime(t.totalMs)}</strong>
            <ProgressBar value={t.progress * 100} max={100} />
          </div>

          <div style={{ display:"flex", gap:8 }}>
            {!t.running && !t.done && <Button onClick={t.start} data-ui="primary-cta">Démarrer</Button>}
            {t.running && <Button onClick={t.pause}>Pause</Button>}
            <Button onClick={() => { t.reset(); }}>Reset</Button>
          </div>
        </section>
      </Card>

      <Card style={{ marginTop: 12 }}>
        <h2>Checklist (max 3)</h2>
        <div style={{ display:"flex", gap: 8, marginBottom: 8 }}>
          <input
            type="text"
            placeholder="Ex: ouvrir le dossier, écrire 3 lignes, envoyer 1 mail"
            value={newTask}
            onChange={(e)=>setNewTask(e.target.value)}
            maxLength={80}
          />
          <Button onClick={addTask}>Ajouter</Button>
        </div>
        <ul style={{ listStyle:"none", padding:0, display:"grid", gap:8 }}>
          {tasks.slice(0,3).map(tk => (
            <li key={tk.id} style={{ display:"flex", justifyContent:"space-between", gap:8, border:"1px solid var(--card)", borderRadius:12, padding:10 }}>
              <label style={{ display:"flex", gap:6, alignItems:"center" }}>
                <input type="checkbox" checked={!!tk.done} onChange={()=>toggleTask(tk.id)} />
                <span style={{ textDecoration: tk.done ? "line-through" : "none" }}>{tk.label}</span>
              </label>
              <Button onClick={()=>delTask(tk.id)}>Supprimer</Button>
            </li>
          ))}
          {!tasks.length && <em>Ajoute 1 à 3 actions simples pour cadrer ta session.</em>}
        </ul>
      </Card>
    </main>
  );
}

function fmtTime(ms: number) {
  const s = Math.round(ms/1000);
  const m = Math.floor(s/60).toString().padStart(2,"0");
  const ss = (s%60).toString().padStart(2,"0");
  return `${m}:${ss}`;
}
