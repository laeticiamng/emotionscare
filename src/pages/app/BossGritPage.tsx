import React, { useState, useEffect } from 'react';
import { Trophy, Clock, CheckCircle, RotateCcw, Target, Zap, Coffee, Footprints } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { Card } from '@/components/ui/card';
import { Progress } from '@/components/ui/progress';
import { toast } from 'sonner';

interface Challenge {
  id: string;
  title: string;
  description: string;
  icon: any;
  duration: number; // en secondes
  category: 'hydration' | 'writing' | 'movement' | 'mindfulness';
  color: string;
}

const BossGritPage = () => {
  const [activeChallenge, setActiveChallenge] = useState<Challenge | null>(null);
  const [timeRemaining, setTimeRemaining] = useState(0);
  const [challengeStarted, setChallengeStarted] = useState(false);
  const [completedToday, setCompletedToday] = useState<string[]>([]);

  const challenges: Challenge[] = [
    {
      id: 'hydrate',
      title: 'Hydratation Flash',
      description: 'Buvez un verre d\'eau lentement',
      icon: Coffee,
      duration: 60,
      category: 'hydration',
      color: 'from-blue-500 to-cyan-500'
    },
    {
      id: 'walk',
      title: 'Marche Express',
      description: '100 pas autour de vous',
      icon: Footprints,
      duration: 120,
      category: 'movement',
      color: 'from-green-500 to-emerald-500'
    },
    {
      id: 'write',
      title: 'Note Rapide',
      description: '√âcrivez 3 choses positives',
      icon: Target,
      duration: 90,
      category: 'writing',
      color: 'from-purple-500 to-violet-500'
    },
    {
      id: 'breathe',
      title: 'Respir Flash',
      description: '5 respirations profondes',
      icon: Zap,
      duration: 45,
      category: 'mindfulness',
      color: 'from-orange-500 to-red-500'
    }
  ];

  // Timer du d√©fi
  useEffect(() => {
    if (!challengeStarted || timeRemaining <= 0) return;

    const timer = setInterval(() => {
      setTimeRemaining(prev => {
        if (prev <= 1) {
          completeChallenge();
          return 0;
        }
        return prev - 1;
      });
    }, 1000);

    return () => clearInterval(timer);
  }, [challengeStarted, timeRemaining]);

  const startChallenge = (challenge: Challenge) => {
    setActiveChallenge(challenge);
    setTimeRemaining(challenge.duration);
    setChallengeStarted(true);
    
    toast.success(`D√©fi "${challenge.title}" lanc√© !`, {
      description: challenge.description,
      duration: 2000
    });
  };

  const completeChallenge = () => {
    if (!activeChallenge) return;

    setChallengeStarted(false);
    setCompletedToday(prev => [...prev, activeChallenge.id]);
    
    toast.success('D√©fi r√©ussi !', {
      description: 'F√©licitations pour cette belle action ! üéâ',
      duration: 3000
    });

    setActiveChallenge(null);
    setTimeRemaining(0);
  };

  const abandonChallenge = () => {
    if (!activeChallenge) return;

    setChallengeStarted(false);
    
    toast.success('D√©j√† un pas fait.', {
      description: 'L\'intention compte aussi ! R√©essayez quand vous voulez.',
      duration: 2000
    });

    setActiveChallenge(null);
    setTimeRemaining(0);
  };

  const formatTime = (seconds: number) => {
    const mins = Math.floor(seconds / 60);
    const secs = seconds % 60;
    return `${mins}:${secs.toString().padStart(2, '0')}`;
  };

  const getChallengeProgress = () => {
    if (!activeChallenge) return 0;
    const elapsed = activeChallenge.duration - timeRemaining;
    return (elapsed / activeChallenge.duration) * 100;
  };

  const getMotivationalMessage = () => {
    const messages = [
      'Vous pouvez le faire !',
      'Un petit pas, un grand effet !',
      'Prendre soin de soi, c\'est important !',
      'Chaque geste compte !',
      'Vous √™tes sur la bonne voie !',
      'Continuez, c\'est parfait !'
    ];
    return messages[Math.floor(Math.random() * messages.length)];
  };

  return (
    <div className="min-h-screen bg-gradient-to-br from-background to-muted p-4">
      <div className="max-w-4xl mx-auto">
        {/* Header */}
        <div className="text-center mb-8">
          <h1 className="text-4xl font-bold bg-gradient-to-r from-primary to-primary/70 bg-clip-text text-transparent mb-2">
            Boss Grit
          </h1>
          <p className="text-muted-foreground">
            Mini-d√©fis bienveillants pour booster votre journ√©e
          </p>
          {completedToday.length > 0 && (
            <p className="text-sm text-primary mt-2">
              üèÜ {completedToday.length} d√©fi{completedToday.length > 1 ? 's' : ''} relev√©{completedToday.length > 1 ? 's' : ''} aujourd'hui !
            </p>
          )}
        </div>

        {activeChallenge ? (
          /* D√©fi actif */
          <div className="space-y-6">
            <Card className="p-8 text-center bg-card/50 backdrop-blur-sm border-muted">
              <div className="space-y-6">
                <div className={`w-20 h-20 mx-auto bg-gradient-to-r ${activeChallenge.color} rounded-full flex items-center justify-center`}>
                  <activeChallenge.icon className="w-10 h-10 text-white" />
                </div>
                
                <div>
                  <h2 className="text-2xl font-bold mb-2">{activeChallenge.title}</h2>
                  <p className="text-muted-foreground mb-4">{activeChallenge.description}</p>
                  
                  {challengeStarted && (
                    <div className="space-y-4">
                      <div className="text-4xl font-bold text-primary">
                        {formatTime(timeRemaining)}
                      </div>
                      <Progress value={getChallengeProgress()} className="w-full max-w-md mx-auto" />
                      <p className="text-sm text-muted-foreground">
                        {getMotivationalMessage()}
                      </p>
                    </div>
                  )}
                </div>

                <div className="flex gap-4 justify-center">
                  {challengeStarted ? (
                    <>
                      <Button onClick={completeChallenge} size="lg">
                        <CheckCircle className="w-5 h-5 mr-2" />
                        Termin√© !
                      </Button>
                      <Button onClick={abandonChallenge} variant="outline" size="lg">
                        <RotateCcw className="w-5 h-5 mr-2" />
                        Arr√™ter
                      </Button>
                    </>
                  ) : (
                    <Button onClick={() => startChallenge(activeChallenge)} size="lg">
                      <Clock className="w-5 h-5 mr-2" />
                      Commencer
                    </Button>
                  )}
                </div>
              </div>
            </Card>
          </div>
        ) : (
          /* S√©lection de d√©fi */
          <div className="space-y-6">
            <div className="text-center">
              <h2 className="text-xl font-semibold mb-4">Choisissez votre d√©fi</h2>
              <p className="text-muted-foreground">
                Des actions simples pour prendre soin de vous
              </p>
            </div>

            <div className="grid md:grid-cols-2 gap-4">
              {challenges.map((challenge) => (
                <Card key={challenge.id} className="p-6 bg-card/30 backdrop-blur-sm border-muted hover:border-primary/50 transition-colors cursor-pointer">
                  <button 
                    onClick={() => startChallenge(challenge)}
                    className="w-full text-left"
                    disabled={completedToday.includes(challenge.id)}
                  >
                    <div className="flex items-start gap-4">
                      <div className={`w-12 h-12 bg-gradient-to-r ${challenge.color} rounded-lg flex items-center justify-center flex-shrink-0`}>
                        <challenge.icon className="w-6 h-6 text-white" />
                      </div>
                      <div className="flex-1">
                        <div className="flex items-center gap-2 mb-2">
                          <h3 className="font-semibold">{challenge.title}</h3>
                          {completedToday.includes(challenge.id) && (
                            <CheckCircle className="w-4 h-4 text-green-500" />
                          )}
                        </div>
                        <p className="text-sm text-muted-foreground mb-2">
                          {challenge.description}
                        </p>
                        <div className="flex items-center gap-2 text-xs text-muted-foreground">
                          <Clock className="w-3 h-3" />
                          <span>{formatTime(challenge.duration)}</span>
                          {completedToday.includes(challenge.id) && (
                            <span className="text-green-600 font-medium ml-2">‚úì Compl√©t√©</span>
                          )}
                        </div>
                      </div>
                    </div>
                  </button>
                </Card>
              ))}
            </div>

            {/* Conseils */}
            <Card className="p-6 bg-card/30 backdrop-blur-sm border-muted">
              <h3 className="font-semibold mb-4 flex items-center gap-2">
                <Trophy className="w-5 h-5 text-primary" />
                Conseils Boss Grit
              </h3>
              <ul className="text-sm text-muted-foreground space-y-2">
                <li>‚Ä¢ Choisissez le d√©fi qui vous parle le plus √† l'instant T</li>
                <li>‚Ä¢ Pas de pression : l'intention compte autant que la r√©alisation</li>
                <li>‚Ä¢ Vous pouvez refaire les m√™mes d√©fis plusieurs fois par jour</li>
                <li>‚Ä¢ C√©l√©brez chaque petite victoire, elle compte vraiment !</li>
                <li>‚Ä¢ En cas d'abandon : pas de jugement, juste de la bienveillance</li>
              </ul>
            </Card>

            {/* B√©n√©fices */}
            <Card className="p-6 bg-card/30 backdrop-blur-sm border-muted">
              <h3 className="font-semibold mb-4">Pourquoi ces micro-d√©fis ?</h3>
              <ul className="text-sm text-muted-foreground space-y-2">
                <li>‚Ä¢ <strong>Hydratation :</strong> R√©veil du corps et de l'esprit</li>
                <li>‚Ä¢ <strong>Mouvement :</strong> Activation douce de l'√©nergie</li>
                <li>‚Ä¢ <strong>√âcriture :</strong> Reconnexion avec le positif</li>
                <li>‚Ä¢ <strong>Respiration :</strong> Ancrage dans l'instant pr√©sent</li>
              </ul>
            </Card>
          </div>
        )}
      </div>
    </div>
  );
};

export default BossGritPage;