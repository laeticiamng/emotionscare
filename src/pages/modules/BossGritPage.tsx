// @ts-nocheck
import React, { useState, useCallback } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { Hammer, Flame, Shield, Sword, Crown } from 'lucide-react';
import { UniverseEngine } from '@/components/universe/UniverseEngine';
import { RewardSystem } from '@/components/rewards/RewardSystem';
import { useRewardsStore } from '@/store/rewards.store';
import { useOptimizedAnimation } from '@/hooks/useOptimizedAnimation';
import { UNIVERSE_CONFIGS } from '@/data/universes/config';
import { Button } from '@/components/ui/button';

interface ForgedItem {
  id: string;
  name: string;
  icon: React.ElementType;
  progress: number;
  description: string;
}

const BossGritPage: React.FC = () => {
  const [isEntering, setIsEntering] = useState(true);
  const [phase, setPhase] = useState<'setup' | 'forging' | 'complete'>('setup');
  const [selectedChallenge, setSelectedChallenge] = useState<string | null>(null);
  const [hammerStrike, setHammerStrike] = useState(0);
  const [showReward, setShowReward] = useState(false);
  
  const { addReward } = useRewardsStore();
  const { entranceVariants, cssAnimationClasses } = useOptimizedAnimation();

  const universe = UNIVERSE_CONFIGS.bossGrit;

  const challenges = [
    { id: 'daily-breath', name: 'Respirer 3 fois par jour', icon: Flame, days: 7 },
    { id: 'journal-week', name: 'Écrire 1 ligne par jour', icon: Shield, days: 5 },
    { id: 'mindful-moments', name: 'Pause mindful quotidienne', icon: Sword, days: 3 }
  ];

  const forgedItems: ForgedItem[] = [
    {
      id: 'flame-sword',
      name: 'Épée de Persévérance',
      icon: Sword,
      progress: 75,
      description: 'Forgée dans la détermination quotidienne'
    },
    {
      id: 'wisdom-shield', 
      name: 'Bouclier de Sagesse',
      icon: Shield,
      progress: 45,
      description: 'Protection contre les doutes'
    }
  ];

  const handleEnterComplete = useCallback(() => {
    setIsEntering(false);
  }, []);

  const selectChallenge = useCallback((challengeId: string) => {
    setSelectedChallenge(challengeId);
    setPhase('forging');
  }, []);

  const strike = useCallback(() => {
    setHammerStrike(prev => {
      const newStrike = prev + 1;
      if (newStrike >= 5) {
        setTimeout(() => completeForging(), 800);
      }
      return newStrike;
    });
  }, []);

  const completeForging = useCallback(() => {
    setPhase('complete');
    
    const challenge = challenges.find(c => c.id === selectedChallenge);
    const itemNames = ['Amulette de Force', 'Clé de Détermination', 'Pièce de Courage'];
    const itemName = itemNames[Math.floor(Math.random() * itemNames.length)];
    
    const reward = addReward({
      type: 'sticker',
      name: itemName,
      description: `Forgé avec ${challenge?.name}`
    });

    setTimeout(() => {
      setShowReward(true);
    }, 1000);
  }, [selectedChallenge, challenges, addReward]);

  if (phase === 'setup') {
    return (
      <UniverseEngine
        universe={universe}
        isEntering={isEntering}
        onEnterComplete={handleEnterComplete}
        enableParticles={true}
        className="bg-gradient-to-br from-slate-900 via-orange-900 to-red-900"
      >
        <div className="min-h-screen flex flex-col items-center justify-center p-6">
          <motion.div
            className="text-center mb-12"
            variants={entranceVariants}
            initial="hidden"
            animate="visible"
          >
            <div className="inline-flex items-center gap-3 mb-4">
              <Hammer className="w-8 h-8 text-orange-400" />
              <h1 className="text-3xl font-light text-white">La Forge Intérieure</h1>
            </div>
            <p className="text-white/70 text-lg max-w-md">
              Forge ta détermination dans les flammes de la persévérance
            </p>
          </motion.div>

          <div className="grid gap-4 max-w-md w-full">
            <h3 className="text-white/80 text-center mb-4">Choisis ton défi à forger</h3>
            {challenges.map((challenge, index) => (
              <motion.div
                key={challenge.id}
                initial={{ opacity: 0, x: -20 }}
                animate={{ opacity: 1, x: 0 }}
                transition={{ delay: 0.5 + index * 0.1 }}
              >
                <Button
                  onClick={() => selectChallenge(challenge.id)}
                  className="w-full bg-gradient-to-r from-orange-500/20 to-red-500/20 hover:from-orange-500/30 hover:to-red-500/30 text-white rounded-2xl p-6 border border-orange-400/30"
                >
                  <challenge.icon className="w-6 h-6 mr-3 text-orange-400" />
                  <div className="text-left">
                    <p className="font-medium">{challenge.name}</p>
                    <p className="text-sm text-white/60">{challenge.days} jours</p>
                  </div>
                </Button>
              </motion.div>
            ))}
          </div>
        </div>
      </UniverseEngine>
    );
  }

  if (phase === 'forging') {
    const challenge = challenges.find(c => c.id === selectedChallenge);
    
    return (
      <UniverseEngine
        universe={universe}
        isEntering={false}
        onEnterComplete={() => {}}
        enableParticles={true}
        className="bg-gradient-to-br from-slate-900 via-orange-900 to-red-900"
      >
        <div className="min-h-screen flex flex-col items-center justify-center p-6">
          <motion.div
            className="text-center mb-8"
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
          >
            <h2 className="text-2xl font-light text-white mb-2">
              Forge: {challenge?.name}
            </h2>
            <p className="text-white/60">
              Frappe {5 - hammerStrike} fois pour terminer
            </p>
          </motion.div>

          {/* Anvil and Hammer */}
          <motion.div
            className="relative mb-12"
            initial={{ scale: 0.8, opacity: 0 }}
            animate={{ scale: 1, opacity: 1 }}
          >
            {/* Sparks effect */}
            <AnimatePresence>
              {Array.from({ length: hammerStrike }).map((_, i) => (
                <motion.div
                  key={i}
                  className="absolute w-2 h-2 bg-orange-400 rounded-full"
                  initial={{ 
                    opacity: 1, 
                    scale: 1,
                    x: Math.random() * 100 - 50,
                    y: Math.random() * 100 - 50
                  }}
                  animate={{ 
                    opacity: 0, 
                    scale: 0,
                    x: Math.random() * 200 - 100,
                    y: Math.random() * 200 - 100
                  }}
                  exit={{ opacity: 0 }}
                  transition={{ duration: 1 }}
                />
              ))}
            </AnimatePresence>

            {/* Anvil representation */}
            <div className="w-32 h-32 bg-gradient-to-b from-gray-300 to-gray-600 rounded-lg relative">
              <div className="absolute inset-2 bg-gradient-to-b from-gray-400 to-gray-500 rounded" />
              <div className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2">
                <Crown className="w-8 h-8 text-orange-400" />
              </div>
            </div>
          </motion.div>

          {/* Strike Button */}
          <Button
            onClick={strike}
            className="w-32 h-32 rounded-full bg-gradient-to-b from-orange-500 to-red-600 hover:from-orange-600 hover:to-red-700 text-white shadow-2xl"
            disabled={hammerStrike >= 5}
          >
            <Hammer className="w-12 h-12" />
          </Button>

          {/* Progress */}
          <div className="mt-8 flex gap-2">
            {Array.from({ length: 5 }).map((_, i) => (
              <div
                key={i}
                className={`w-3 h-3 rounded-full transition-all duration-300 ${
                  i < hammerStrike ? 'bg-orange-400' : 'bg-white/20'
                }`}
              />
            ))}
          </div>
        </div>
      </UniverseEngine>
    );
  }

  return (
    <UniverseEngine
      universe={universe}
      isEntering={false}
      onEnterComplete={() => {}}
      enableParticles={true}
      className="bg-gradient-to-br from-slate-900 via-orange-900 to-red-900"
    >
      <div className="min-h-screen flex flex-col items-center justify-center p-6">
        <motion.div
          className="text-center mb-8"
          initial={{ opacity: 0, y: 20 }}
          animate={{ opacity: 1, y: 0 }}
        >
          <div className="inline-flex items-center gap-3 mb-4">
            <div className={`w-12 h-12 bg-orange-400 rounded-full flex items-center justify-center ${cssAnimationClasses.pulse}`}>
              <Crown className="w-6 h-6 text-white" />
            </div>
            <h2 className="text-2xl font-light text-white">Artefact forgé</h2>
          </div>
          <p className="text-white/70">
            Ton objet de détermination est prêt
          </p>
        </motion.div>

        <div className="max-w-md w-full mb-8">
          <h3 className="text-white/80 text-center mb-4">Ta collection</h3>
          <div className="space-y-3">
            {forgedItems.map((item) => (
              <div
                key={item.id}
                className="bg-white/10 backdrop-blur-md rounded-xl p-4 border border-orange-400/20"
              >
                <div className="flex items-center gap-3 mb-2">
                  <item.icon className="w-5 h-5 text-orange-400" />
                  <span className="text-white font-medium">{item.name}</span>
                </div>
                <div className="w-full bg-white/20 rounded-full h-2 mb-2">
                  <div 
                    className="bg-gradient-to-r from-orange-400 to-red-400 h-2 rounded-full transition-all duration-500"
                    style={{ width: `${item.progress}%` }}
                  />
                </div>
                <p className="text-white/60 text-sm">{item.description}</p>
              </div>
            ))}
          </div>
        </div>

        {showReward && (
          <RewardSystem
            type="sticker"
            message="Objet forgé avec persévérance"
            onComplete={() => setShowReward(false)}
          />
        )}
      </div>
    </UniverseEngine>
  );
};

export default BossGritPage;