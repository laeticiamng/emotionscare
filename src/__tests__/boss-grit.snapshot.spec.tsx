import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
vi.mock("@/components/ui/PageHeader", () => ({
  default: ({ title }: any) => <div>{title}</div>,
}));

vi.mock("@/components/ui/card", () => ({
  Card: ({ children }: any) => <div>{children}</div>,
}));

vi.mock("@/components/ui/button", () => ({
  Button: ({ children, ...props }: any) => <button {...props}>{children}</button>,
}));

vi.mock("@/ui/ProgressBar", () => ({
  ProgressBar: ({ value }: any) => <div data-progress={value} />,
}));
import BossGritPage from "@/modules/boss-grit/BossGritPage";

describe("BossGritPage", () => {
  it("rend la page et le bouton Démarrer", () => {
    const { container, getByText } = render(<BossGritPage />);
    expect(getByText(/Boss Grit/i)).toBeTruthy();
    expect(getByText(/Démarrer/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
