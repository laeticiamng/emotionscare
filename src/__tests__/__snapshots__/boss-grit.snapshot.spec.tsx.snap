// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BossGrit Page > rend la page 1`] = `
<div>
  <main
    class="p-4 space-y-4"
  >
    <div>
      BossGrit
    </div>
    <p>
      Bienvenue dans le module BossGrit.
    </p>
    <button>
      Commencer
    </button>
  </main>
</div>
`;
